pkg/api/validation/validation.go

```diff
@@ -2254,6 +2276,21 @@ func ValidatePod(pod *api.Pod) field.ErrorList {
 	allErrs = append(allErrs, validateContainersOnlyForPod(pod.Spec.Containers, specPath.Child("containers"))...)
 	allErrs = append(allErrs, validateContainersOnlyForPod(pod.Spec.InitContainers, specPath.Child("initContainers"))...)
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.HugePages) {
+		hugePageResources := sets.NewString()
+		for i := range pod.Spec.Containers {
+			resourceSet := toContainerResourcesSet(&pod.Spec.Containers[i])
+			for resourceStr := range resourceSet {
+				if v1helper.IsHugePageResourceName(v1.ResourceName(resourceStr)) {
```

Comment:
@xiangpengzhao - that is a good question.

there are a variety of large page sizes depending on the architecture, so i was hoping to avoid an explicit enumeration of them or awareness in the api server.  this is an alpha feature, but my long term expectation is that operators would configure the `ResourceQuota` system to make consumption of hugepages denied by default per something like https://github.com/kubernetes/kubernetes/pull/36765 , and they would then grant explicit quota for the page sizes supported in their fleet of machines.  WDYT?